server/src/main/java/org/jboss/as/server/deployment/module/ModuleAliasChecker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.server.deployment.module;
+
+import java.util.List;
+
+import org.jboss.as.server.logging.ServerLogger;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleLoadException;
+import org.jboss.modules.ModuleLoader;
+
+/**
+ * An internal utility class to do common tasks related to check module alias when they are supplied as deployment dependencies.
+ */
+public class ModuleAliasChecker {
+
+    public enum MessageContext {
+        JBOSS_DEPLOYMENT_STRUCTURE_CONTEXT("jboss-deployment-structure.xml"), MANIFEST_CONTEXT("MANIFEST.MF");
+
+        private final String context;
+
+        MessageContext(String context) {
+            this.context = context;
+        }
+
+        @Override
+        public String toString() {
+            return context;
+        }
+    }
+
+    /**
+     * Returns the target module of a given module identifier if the identifier represents a module alias.
+     *
+     * @param identifier A module identifier
+     * @return The target module identifier of this alias or {@code null} if this identifier does not represent an alias.
+     */
+    static String getTargetModule(String identifier) {
+        final ModuleLoader moduleLoader = Module.getBootModuleLoader();
+        try {
+            Module module = moduleLoader.loadModule(identifier);
+            String moduleName = module.getName();
+            if (!moduleName.equals(identifier)) {
+                return moduleName;
+            }
+        } catch (ModuleLoadException e) {
+            // Explicitly ignored, we are not interested in knowing whether the module cannot be loaded at this point.
+        }
```

Comment:
I think we should wrap L53-62 in a 'if (identifier.startsWith("javax.")) {' block. Just to be conservative at the start.

Reply:
if (identifier.startsWith("javax." && identifier.contains(".api")) {

Reply:
That will be ok, but we need to be sure all the work replacing DUPs only affect to javax modules

Reply:
@bstansberry done